core/src/main/java/org/frankframework/jdbc/JdbcTransactionalStorage.java

```diff
@@ -566,122 +542,64 @@ protected String storeMessageInDatabase(Connection conn, String messageId, Strin
-						return "<id>" + rs.getString(1) + "</id>";
-					}
-				}
-
-				boolean isMessageDifferent = isMessageDifferent(conn, messageId, message);
-				String resultString = createResultString(isMessageDifferent);
-				log.warn("MessageID [{}] already exists", messageId);
-				if (isMessageDifferent) {
-					log.warn("Message with MessageID [{}] is not equal", messageId);
-				}
-				return resultString;
-			}
-			if (isOnlyStoreWhenMessageIdUnique()) {
-				stmt.setString(++parPos, messageId);
-				stmt.setString(++parPos, getSlotId());
 			}
+
 			stmt.execute();
-			updateCount = stmt.getUpdateCount();
-			if (log.isDebugEnabled()) {
-				log.debug("update count for insert statement: " + updateCount);
-			}
-		}
-		if (updateCount > 0) {
-			if (log.isDebugEnabled()) {
-				log.debug("preparing select statement [{}]", selectKeyQuery);
-			}
-			// retrieve the key
-			String newKey;
-			try (PreparedStatement stmt = conn.prepareStatement(selectKeyQuery); ResultSet rs = stmt.executeQuery()) {
-				if (!rs.next()) {
-					throw new SenderException("could not retrieve key of stored message");
+			try (ResultSet rs = stmt.getGeneratedKeys()) {
+				if (rs.next() && rs.getString(1) != null) {
+					return "<id>" + rs.getString(1) + "</id>";
 				}
-				newKey = rs.getString(1);
 			}
+			log.warn("No keys generated after INSERT statement");
 
-			// and update the blob
-			if (log.isDebugEnabled()) {
-				log.debug("preparing update statement [{}]", updateBlobQuery);
-			}
-			try (PreparedStatement stmt = conn.prepareStatement(updateBlobQuery);){
-				stmt.clearParameters();
-				stmt.setString(1, newKey);
+			return null;
+		}
+	}
 
-				try (ResultSet rs = stmt.executeQuery()) {
-					if (!rs.next()) {
-						throw new SenderException("could not retrieve row for stored message [" + messageId + "]");
-					}
-					Object blobHandle = dbmsSupport.getBlobHandle(rs, 1);
-					try (ObjectOutputStream oos = new ObjectOutputStream(JdbcUtil.getBlobOutputStream(dbmsSupport, blobHandle, rs, 1, isBlobsCompressed()))) {
-						oos.writeObject(message);
+	private String checkIfMessageIdAlreadyStored(Connection conn, String messageId, S message) throws SQLException {
+		log.debug("Preparing select key statement [{}]", selectKeyForMessageQuery);
+		try (PreparedStatement stmt = conn.prepareStatement(selectKeyForMessageQuery)) {
+			stmt.setString(1, getSlotId());
+			stmt.setString(2, messageId);
+			try (ResultSet rs = stmt.executeQuery()) {
+				if (rs.next()) {
+					boolean isMessageDifferent = isStoreFullMessage() && isMessageDifferent(rs, 2, messageId, message);
+					String resultString = createWarningResultString(isMessageDifferent);
+					log.warn("MessageID [{}] already exists", messageId);
+					if (isMessageDifferent) {
+						log.warn("Message with MessageID [{}] is not equal", messageId);
 					}
-					dbmsSupport.updateBlob(rs, 1, blobHandle);
-					return "<id>" + newKey + "</id>";
+					return resultString;
```

Comment:
I've considered adding the key of the existing message to the result here, but that will change the XML making it potentially incompatible.

I think it might be useful though to have that key.

So I'm not sure what to do, open for discussion / suggestions.

Reply:
Ik denk dat men een xpath gebruikt en dus dat het toevoegen van een extra element geen kwaad kan!

Reply:
Ik zal dit extra element nog toevoegen.